frontend/public/components/secrets/create-secret/PullSecretCredentialEntry.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Base64 } from 'js-base64';
+import { useEffect, useState } from 'react';
+
+export const PullSecretCredentialEntry: React.FC<PullSecretCredentialEntryProps> = ({
+  id,
+  uid,
+  entry,
+  onChange,
+}) => {
+  const { t } = useTranslation();
+  const [state, setState] = useState<PullSecretCredentialEntryState>({
+    address: entry?.address ?? '',
+    username: entry?.username ?? '',
+    password: entry?.password ?? '',
+    email: entry?.email ?? '',
+    auth: entry?.auth ?? '',
+    uid: uid ?? '',
+  });
+
+  useEffect(() => {
+    onChange(state, id);
+  }, [state, id, onChange]);
+
+  const handleChange = (field: keyof PullSecretCredentialEntryState) => (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const value = event.currentTarget.value;
+    setState((prevState) => {
+      const newState = {
+        ...prevState,
+        [field]: value,
+      };
+      if (field === 'username' || field === 'password') {
+        newState.auth = Base64.encode(`${newState.username}:${newState.password}`);
+      }
+      return newState;
+    });
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    const { name, value } = event.currentTarget;
+    setState((prevState) => ({
+      ...prevState,
+      [name]: value.trim(),
+    }));
+  };
```

Comment:
We only need to update the parent 'entry' whenever a text field changes. This logic could be simplified to:
``` suggestion
  const udpateEntry = (name, value) =>
    onChange({ address, username, password, email, [name]: value });
  
  const handleBlurEvent = (e: React.SyntheticEvent<HTMLInputElement>) =>
    updateEntry(e.currentTarget.name, e.currentTarget.value.trim());
  
  const handleChangeEvent = (e: React.SyntheticEvent<HTMLInputElement>) =>
    updateEntry(e.currentTarget.name, e.currentTarget.value);
```